1st Deliverable

25eightpunkweight opened the pull request.

Commit:
deliv-1 initial commit

Commit:
deliv-1 importing Header in App

Commit:
deliv-1 header component export

Commit:
deliv-1 uuuhhhh

Commit:
deliv-1 sdfasfa

Commit:
deliv-1 bruh

Commit:
deliv-1 bruh ok

Commit:
deliv-1 bruh ok wait I should change the route

Commit:
deliv-1 bruh ok wait I should change the route

Commit:
deliv-1 bruh css

Commit:
deliv-1 sssadasd

Commit:
deliv-1 adasdasfsf

Commit:
deliv-1 trying out text box in header

Commit:
deliv-1 disable text field

Commit:
deliv-1 make the header bigger?

Commit:
deliv-1 make the header bigger?

Commit:
deliv-1 make the header bigger?

Commit:
deliv-1 make the header bigger?

Commit:
deliv-1 make the header bigger?

Commit:
deliv-1 give the header a button

Commit:
deliv-1 add a login/logout button

25eightpunkweight committed:
Add files via upload

Commit:
deliv-1 add mock json

Commit:
deliv-1 ok bruh ok carditems letsgo

Commit:
deliv-1 boop

Commit:
deliv-1 boop2

Commit:
deliv-1 boop2

Commit:
deliv-1 somethingboop

Commit:
deliv-1 somethingboop1

Commit:
deliv-1 ewan

Commit:
deliv-1 amogus

Commit:
deliv-1 it's raining 15:09

Commit:
deliv-1 15:11 jun4-2021

Commit:
deliv-1 15:13 jun4-2021

Commit:
deliv-1 ok

Commit:
deliv-1

Commit:
deliv-1 hasanpiker

Commit:
deliv-1 boooop

Commit:
deliv-1 ok what about this

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-11

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1 dreamvshasan

Commit:
deliv-1 stop

Commit:
deliv-1

Commit:
please

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
f

Commit:
bodycomponent

Commit:
deliv-1

Commit:
deliv

Commit:
deliv-1

Commit:
deliv-1 learn to use git add -u dipshit

Commit:
deliv-1

Commit:
deliv-1 uuummm

Commit:
deliv-1 buttons

Commit:
deliv-1 button colors

Commit:
deliv-1 array manipulation

Commit:
deliv-1 card item styling

Commit:
deliv-1 styling for cards div

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

Commit:
deliv-1

tamnguyenwizeline reviewed (commented):
![image](https://user-images.githubusercontent.com/62417627/121777353-cb982200-cbbb-11eb-90e3-60fe5bee3410.png)
sorry for the late reply! can you please check your code works correctly first?

tamnguyenwizeline commented on public/youtube-videos-mock.json:
```diff
@@ -0,0 +1,856 @@
+{
```
you shouldn't put this mock file into public folder

tamnguyenwizeline commented on src/pages/Home/Home.page.jsx:
```diff
@@ -3,11 +3,15 @@ import { Link, useHistory } from 'react-router-dom';
   const sectionRef = useRef(null);
-  const { authenticated, logout } = useAuth();
+  // const { authenticated, logout } = useAuth();
+  const trimmedArray = mockData.items;//.slice(0, 5);
```
the name should be meaningful for example `videos`

tamnguyenwizeline reviewed (commented):
hi @25eightpunkweight  please take a look on those comments, and also your code is missing styled-component implementation

25eightpunkweight committed:
deliv-1 use styled components!

25eightpunkweight committed:
deliv-1 using styled-componentes

25eightpunkweight committed:
deliv-1 fix props referencing

25eightpunkweight committed:
deliv-1 fixes

25eightpunkweight committed:
deliv-1 fix

25eightpunkweight committed:
deliv-1 fixes

25eightpunkweight committed:
deliv-1 fixes

25eightpunkweight committed:
deliv-1 fixes

25eightpunkweight committed:
deliv-1 fixes

25eightpunkweight committed:
deliv-1 using styled-components on the header

25eightpunkweight committed:
deliv-1 styling header

25eightpunkweight committed:
deliv-1 styled-component header

25eightpunkweight committed:
styled-components on Body

tamnguyenwizeline reviewed (commented):
hi @25eightpunkweight, thanks for your update. In general, your code looks better now. Hopefully, you get the basic idea of styled-component and go beyond with that. I also wanna introduce you another library often used together with styled-component is https://styled-system.com/

tamnguyenwizeline commented:
![Uploading image.png…]()
the first row UI is looked weird

qtd-wizeline commented on src/components/Body/Body.component.jsx:
```diff
@@ -0,0 +1,23 @@
+function Body() {
+  return (
+    <Layout>
+      <Switch>
```
Should put `Switch` outside of `Layout` in case we have different layouts for different routes, this is for later scaling

25eightpunkweight replied:
should it be a `Layout` tag before showing the component of each `Route`?

> `<Switch>`
> `  <Route>`
> `    <Layout>`
> `      <Component />`

qtd-wizeline commented on src/components/App/App.component.jsx:
```diff
@@ -32,25 +34,8 @@ function App() {
-          <Fortune />
-        </Layout>
-      </AuthProvider>
+      <Header />
```
We can put `Header` and `Body` inside `Layout`

25eightpunkweight replied:
but body contains `Layout`.

25eightpunkweight replied:
can you please give me more details?

25eightpunkweight replied:
but wouldn't `Body` already loop with that? or do you also mean to redesign `Body` completely?

qtd-wizeline commented on package.json:
```diff
@@ -47,15 +48,9 @@
-      "pretty-quick --staged",
-      "git add"
-    ]
+    "*.{js, jsx, css, json}": []
```
Please re-enable linting for code formatting, this keep out source code consistent and easy to maintain

qtd-wizeline commented on src/components/CardItem/CardItem.component.jsx:
```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components';
+
+// import { useFortune } from '../../utils/hooks/useFortune';
+// import './CardItem.styles.css';
```
Please remove if we don't need these lines of code, for readability

qtd-wizeline commented on src/components/CardItem/CardItem.component.jsx:
```diff
@@ -0,0 +1,41 @@
+  position: relative;
+`;
+
+function Component(item) {
```
Should name the component the same with file name, `CardItem` in this case

qtd-wizeline commented on src/pages/Home/Home.page.jsx:
```diff
@@ -1,23 +1,16 @@
-import { Link, useHistory } from 'react-router-dom';
 
-import { useAuth } from '../../providers/Auth';
 import './Home.styles.css';
```
Should use only `styled-component` for styling, combining both could be cumbersome

qtd-wizeline commented on src/components/CardItem/CardItem.styles.css:
```diff
@@ -0,0 +1,80 @@
+.card-title {
+  margin: 1px;
+  height: 56px;
+}
```
Please remove redundant file

25eightpunkweight committed:
deliv-1 match component exports with their filenames

 + organizing components in Body for scalability

 + enable linting + remove unnecessary declarations in App component

 + remove unused css

 + use styled-components on the pages + delete unuesed styles

 + adjust css
